byway/src/main/webapp/routepage.js

```diff
@@ -49,27 +64,135 @@ function calcRoute(directionsService, directionsRenderer, start, end) {
   });
 }
 
+/** Get the new list of stops from datastore onload */
+function getStops(){
+  const stopList = document.getElementById('stop-list');
+  if(stopList != null){
+        stopList.innerText = ""; // clear list
+    }
+  fetch('/api/stop')
+  .then(response => response.json())
+  .then((stops) => {
+    stops.forEach((stop)=>{
+      var btn = document.createElement('button');
+      btn.id = stop.id;
+      btn.innerText = stop.placename;
+      btn.setAttribute("class", "btn rec-btn");
+      btn.addEventListener("click", function() {
+        deleteFromStops(stop); 
+      });
+     stopList.appendChild(btn);
+    })
+  })
+}
+
+/** Get the new list of stops locally */
+function getStopsList(){
+  const stopList = document.getElementById('stop-list');
+  if(stopList != null){
+    // clear list
+    stopList.innerText = ""; 
+  }
```

Comment:
When would stopList be null? Similar to a comment I left below in the `script.js`, consider only deleting the stop you have to delete rather than clearing them all and loading `n-1` elements back from the server.